### PR TITLE
Added a Makefile that runs the experiments and reproduces the plots.

### PR DESCRIPTION
**Pull request by ioreshnikov**
This make `experiments.sh` obsolete, so I removed it. README is updated accordingly.

Note: The experiments are being run in a slightly magical way -- by generating a separate makefile target for each output data file. This target depends on the config file. This has three main advantages:

1. We can always be sure that we have all the data files (none of them missing) and they are up-to-date (if you update the config file the data files will be rebuilt).
2. We never rebuild the datafiles if the corresponding experiment has been run already.
3. We can use make's `-j` flag to run the experiments in multitple parallel processes. I've tried it on my machine, and I can comfortably run 4 experiments at a time.

The disadvantage, of course, is that it's quite difficult to read the makefile :)

### CODE REVIEW

**Comment by ioreshnikov** [comment]
@jcpassy can you please take a look at the Makefile?

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# This target creates a new virtual environment and then installs all
+# the required packages in there.
+venv:
+	@echo "Building a venv"
+	python -m venv venv
+	. venv/bin/activate
+	python -m pip install -r requirements.txt
```
Comment (jcpassy) [inline Makefile:7]: I am against delegating the responsibility to build a venv to the makefile. First, the user technically does not have to. It is just good practice. Secondly, the user completely loses control on the venv (using --copies or not, naming the venv, using a preexisting venv, even using conda...).
I would remove anything related to venv in the makefile and add these as suggestions in the README. I know it was done in the mesh package but I removed it. :)

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+# The experiment is defined by its number. Here we generate all the
+# experiment numbers and store them in a variable.
+EXP_NUMS = $(shell seq 0 30)
+
+# This macro generates a target named results_N.data.
+define make-experiment-result-target
+$(addsuffix .data, $(addprefix results_, $1)): venv experiments/config.yml
+	. venv/bin/activate
+	cd experiments && PYTHONPATH=.. python experiment.py --exp_number=$(strip $1)
```
Comment (jcpassy) [inline Makefile:27]: If you want to do it like that, I think we should really have an install target that installs the package properly and not mess with the PYTHONPATH. :)
I have the setup.py on my branch so you can just prepare everything here and it should work after the rebase.

**Review by jcpassy — COMMENTED** [review]
Nice, I only have two comments about the logic for creating virtual envs and installing the package.

File: Makefile (hunk 1 above)
Comment (ioreshnikov) [reply Makefile:7]: > the user technically does not have to

I wish they had :( But fair point, I'll remove this target.

File: Makefile (hunk 2 above)
Comment (ioreshnikov) [reply Makefile:27]: I'll rebase on top of yours. Otherwise you'd have to debug my makefile :)

**Pull request merged**